apollo-federation/src/sources/connect/validation/mod.rs

```diff
@@ -73,18 +73,40 @@ use crate::subgraph::spec::CONTEXT_DIRECTIVE_NAME;
 use crate::subgraph::spec::EXTERNAL_DIRECTIVE_NAME;
 use crate::subgraph::spec::FROM_CONTEXT_DIRECTIVE_NAME;
 
+// The result of a validation pass on a subgraph
+#[derive(Debug)]
+pub struct ValidationResult {
+    /// All validation errors encountered.
+    pub errors: Vec<Message>,
+
+    /// Whether or not the validated subgraph contained connector directives
+    pub has_connectors: bool,
+
+    /// The original subgraph name
+    pub subgraph_name: String,
+
+    /// The parsed (and potentially invalid) schema of the subgraph
+    pub schema: Schema,
+}
+
 /// Validate the connectors-related directives `@source` and `@connect`.
 ///
 /// This function attempts to collect as many validation errors as possible, so it does not bail
 /// out as soon as it encounters one.
-pub fn validate(source_text: &str, file_name: &str) -> Vec<Message> {
+pub fn validate(source_text: &str, file_name: &str) -> ValidationResult {
     // TODO: Use parse_and_validate (adding in directives as needed)
     // TODO: Handle schema errors rather than relying on JavaScript to catch it later
     let schema = Schema::parse(source_text, file_name)
         .unwrap_or_else(|schema_with_errors| schema_with_errors.partial);
     let connect_identity = ConnectSpec::identity();
     let Some((link, link_directive)) = Link::for_identity(&schema, &connect_identity) else {
-        return Vec::new(); // There are no connectors-related directives to validate
+        // There are no connectors-related directives to validate
+        return ValidationResult {
+            errors: Vec::new(),
+            has_connectors: false,
+            subgraph_name: file_name.to_string(),
```

Comment:
This isn't necessarily true—I feel like the consumer is the only one who knows the subgraph name and they should use their own context as the source of truth

Reply:
Makes sense. I'll remove this field.